Improve igenomes compatibility

nictru opened the pull request:
Currently the pipeline fails if there are no entries of type `gene` in the GTF file, which is the case for some of the most commonly used igenomes entries. These GTF files only contain CDS and exon entries with correct `gene_id` and `gene_name` fields, but not separate rows for the genes.

This was fixed by using [this module](https://nf-co.re/modules/agat_convertspgxf2gxf/). It adds adds the missing rows based on the existing information.

Additionally I added a check that makes sure that at least 90% of the genes in the provided count matrix are available in the `gene_id` or the `gene_name` fields of the input GTF file. This prevents problems later on. If we cannot make a connection between the genes in the count matrix and their genomic position, they are not useful in this pipeline.

nictru committed:
Improve igenomes compatibility

nictru committed:
Fix one outdir definition

nictru committed:
Improve error message

LeonHafner reviewed (approved):
Thanks for the fix. Look good to me. Feel free to go ahead and merge it once the CI checks pass.

nictru committed:
Fix test data definition

nictru committed:
Update test snapshots

nictru committed:
Update pipeline-level test snapshots

nictru commented:
The GTF and Fasta files in our test data were subsetted to a single chromosome, but our count matrices were not. This triggered the count-matrix - GTF match check to fail.
I now removed the extra genes from the count test data. 

I also removed some unnecessary TF-ChIP-Seq bam files from the BAM samplesheet, as these were downloaded in every test run, but not used anywhere in the pipeline.
